### DISCUSSION TITLE
org.gradle.api.publication.maven.internal.DefaultMavenFactory

### DISCUSSION THREAD

**Discussion by c1p81** [apache/cordova#367] [Q&A]
Recompiling an old project with Android Studio 2021.3.1 Patch 1 i have this message

Unable to load class 'org.gradle.api.publication.maven.internal.MavenPomMetaInfoProvider'.

I have found the same problem in some posts on Stack Overflow but the solutions era not relative to Cordoba

thanks
L

**Comment by breautek** [comment]
There is a rather large chain of dependencies when it comes to Android Studio and the project that cordova produces. So if you have an older project, it's likely not compatible with more recent Android Studio versions. You'll still be able to compile without Android Studio using the cordova cli commands (e.g. `cordova build android`) which is what I'd suggest to do rather than opening up the project in Android Studio.

Android Studio depends on specific ranges of AGP (Android Gradle Plugin), and which often requires a specific gradle version. The cordova project also depends on both of these but might not be compatible with what Android Studio requires.

Since you're dealing with an old project, your options are to:

A) Use an older/previous release of Android Studio that would be compatible with the cordova-android version you're using. We do not keep track of Android Studio compatibility so this might be hard... trial/error...

B) Update cordova-android@11 which should be compatible with the latest version of Android Studio.

C) Don't use Android Studio, simply use the cordova command line to do your builds.

I assume you have opened up the project in Android Studio, and in doing so Android Studio may have updated dependencies that could break the cordova project. So before attempting Option A or C, you may have to reconstruct the project by removing and re-adding the android platform:

```
cordova platform remove android
cordova platform add android@<whatever version you're using>
```
